patroni/postgresql/bootstrap.py

```diff
@@ -48,7 +86,10 @@ def option_is_allowed(name):
                     if len(keys) != 1 or not isinstance(opt[keys[0]], str) or not option_is_allowed(keys[0]):
                         error_handler('Error when parsing {0} key-value option {1}: only one key-value is allowed'
                                       ' and value should be a string'.format(tool, opt[keys[0]]))
-                    user_options.append('--{0}={1}'.format(keys[0], opt[keys[0]]))
+                    try:
+                        user_options.append('--{0}={1}'.format(keys[0], shell_quote(opt[keys[0]])))
+                    except ValueError as err:
```

Comment:
IMO this exception should be handled in the `is_single_quoted_string()` function by simply returning a `False`.